sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileIO.java

```diff
@@ -426,6 +426,28 @@ public String readFullyAsUTF8String() throws IOException {
     public String toString() {
       return "ReadableFile{metadata=" + metadata + ", compression=" + compression + '}';
     }
+
+    @Override public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      ReadableFile that = (ReadableFile) o;
+
+      if (metadata != null ? !metadata.equals(that.metadata) : that.metadata != null) {
```

Comment:
This is causing the findbugs failures;

I would go something like

```
if (this == o) { return true; } else if (!(o instanceof ReadableFile)) { return false; }
...
return Objects.equals(this.metadata, that.metadata) && Objects.equals(this.compression, that.compression);
```

You may also need to annotate the Object with `@Nullable`, because I think we've propagated "parametersAreNonNullByDefault", which may be throwing off findbugs based on the expected equals contract.

Reply:
Trying guava. I used intellij generated equals/hashCode. Weird